clang/lib/CodeGen/CGCall.cpp

```diff
@@ -5085,7 +5085,7 @@ RValue CodeGenFunction::EmitCall(const CGFunctionInfo &CallInfo,
   RawAddress SRetAlloca = RawAddress::invalid();
   llvm::Value *UnusedReturnSizePtr = nullptr;
   if (RetAI.isIndirect() || RetAI.isInAlloca() || RetAI.isCoerceAndExpand()) {
-    if (IsVirtualFunctionPointerThunk && RetAI.isIndirect()) {
+    if ((IsVirtualFunctionPointerThunk && RetAI.isIndirect()) || IsMustTail) {
```

Comment:
Do any of the tests cover the case of `RetAI.isInAlloca()` or `RetAI.isCoerceAndExpand()`?

Reply:
Waiting for an answer here.

Reply:
Apologies for slow replies on both of my PRs, our fuzzer and some of our downstream testing identified some issues and I had to verify they were unrelated. No none of the tests cover these cases. I am in the process of trying to add some that do.